src/d3a/d3a_core/simulation.py

```diff
@@ -274,6 +274,7 @@ def _execute_simulation(self, slot_resume, tick_resume, console=None):
         slot_count = int(config.sim_duration / config.slot_length)
 
         self._update_and_send_results()
+        self.simulation_config.external_redis_communicator.start_communication()
```

Comment:
Please move this before the update_and_send_results, otherwise the first results transmission will fail.

Reply:
Thanks doing it